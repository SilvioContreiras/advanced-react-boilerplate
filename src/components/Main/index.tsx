const Main = () => (
  <div>
    <h1>React Next</h1>
  </div>
);

export default Main;
