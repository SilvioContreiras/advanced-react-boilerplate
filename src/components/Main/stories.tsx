import { withKnobs, text } from '@storybook/addon-knobs';
import { Meta, Story } from '@storybook/react/types-6-0';

import Main from '.';

export default {
  title: 'Main',
  component: Main,
  decorators: [withKnobs]
} as Meta;

export const Basic: Story = () => (
  <Main
    title={text('Title', 'React AvanÃ§ado')}
    description={text('Description', 'Reactjs, TypeScript, NextJS')}
  />
);
