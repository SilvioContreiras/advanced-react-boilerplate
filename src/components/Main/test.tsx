import { render, screen } from '@testing-library/react';

import Main from '.';

describe('<Main />', () => {
  it('should render heading', () => {
    const { container } = render(<Main />);

    expect(
      screen.getByRole('heading', { name: /react next/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
