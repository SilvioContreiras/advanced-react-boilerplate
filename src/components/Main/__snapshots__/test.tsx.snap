// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render heading 1`] = `
<div>
  <h1>
    React Next
  </h1>
</div>
`;
